services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -1790,6 +1791,87 @@ class HakemusServiceITest(
                 alluClient.getApplicationInformation(alluId)
             }
         }
+
+        @Test
+        fun `kaivuilmoitus with new johtoselvitys works`() {
```

Comment:
This might be testing too many things at once. Could this be broken down into a couple of tests? Like:
- `saves johtoselvitys to DB when kaivuilmoitus requests a new johtoselvitys`
- `sends johtoselvitys to Allu when kaivuilmoitus requests a new johtoselvitys`
- `returns the created kaivuilmoitus when kaivuilmoitus requests a new johtoselvitys`

Maybe the setup could be at least partially shared between the tests? Maybe a nested class for these? With a nested class, the test names might not have to repeat the context.

Reply:
You're right. I split the tests into nested tests. Not sure if the setUp is really clear though.